scripts/MAZE-EN/c100206005.lua

```diff
@@ -0,0 +1,64 @@
+--Gate Guardian of Wind and Water
+--scripted by JoyJ
+function c100206005.initial_effect(c)
+	--fusion material
+	c:EnableReviveLimit()
+	aux.AddFusionProcCode2(c,62340868,98434877,true,true)
+	aux.AddContactFusionProcedure(c,Card.IsAbleToRemoveAsCost,LOCATION_ONFIELD+LOCATION_HAND+LOCATION_REMOVED,0,Duel.Remove,POS_FACEUP,REASON_COST)
+	--disable
+	local e1=Effect.CreateEffect(c)
+	e1:SetDescription(aux.Stringid(100206005,1))
+	e1:SetCategory(CATEGORY_DISABLE)
+	e1:SetType(EFFECT_TYPE_QUICK_O)
+	e1:SetCode(EVENT_CHAINING)
+	e1:SetCountLimit(2,100206005)
+	e1:SetRange(LOCATION_MZONE)
+	e1:SetCondition(c100206005.discon)
+	e1:SetTarget(c100206005.distg)
+	e1:SetOperation(c100206005.disop)
+	c:RegisterEffect(e1)
+	--special summon
+	local e2=Effect.CreateEffect(c)
+	e2:SetDescription(aux.Stringid(100206005,2))
+	e2:SetCategory(CATEGORY_SPECIAL_SUMMON)
+	e2:SetType(EFFECT_TYPE_SINGLE+EFFECT_TYPE_TRIGGER_O)
+	e2:SetCode(EVENT_LEAVE_FIELD)
+	e2:SetProperty(EFFECT_FLAG_DELAY+EFFECT_FLAG_DAMAGE_STEP)
+	e2:SetCondition(c100206005.spcon)
+	e2:SetTarget(c100206005.sptg)
+	e2:SetOperation(c100206005.spop)
+	c:RegisterEffect(e2)
+end
+function c100206005.discon(e,tp,eg,ep,ev,re,r,rp)
+	return ep~=tp and re:IsActiveType(TYPE_SPELL+TYPE_TRAP)
+		and (re:GetActivateLocation() & LOCATION_ONFIELD)>0 and Duel.IsChainDisablable(ev)
+end
+function c100206005.distg(e,tp,eg,ep,ev,re,r,rp,chk)
+	if chk==0 then return c:GetFlagEffect(100206005)==0 end
```

Comment:
redtext: `attempt to index a nil value (global 'c')`
```suggestion
	if chk==0 then return e:GetHandler():GetFlagEffect(100206005)==0 end
```